File: 1781-sum-of-beauty-of-all-substrings/1781-sum-of-beauty-of-all-substrings.java
class Solution {
    public int beautySum(String s) {

        int length = s.length();
        int result = 0;
        int freq[] = new int[26];
        int maxVal = Integer.MIN_VALUE;
        int minVal = Integer.MAX_VALUE;

        for(int i = 0 ; i < length ; i++)   
        {
            for(int j = i ; j < length ; j++)
            {
                freq[s.charAt(j) - 'a']++;
                
                for(int k = 0 ; k < 26 ; k++)
                {
                    if(freq[k] != 0)
                    {
                        maxVal = Math.max(maxVal, freq[k]);
                        minVal = Math.min(minVal, freq[k]);
                    }
                }

                result += maxVal - minVal;
                maxVal = Integer.MIN_VALUE;
                minVal = Integer.MAX_VALUE;
            }

            Arrays.fill(freq,0);
        }

        return result;
    }
}

File: 1781-sum-of-beauty-of-all-substrings/1781-sum-of-beauty-of-all-substrings.js
/**
 * @param {string} s
 * @return {number}
 */
var beautySum = function(s) {
    let length = s.length;
    let maxVal = Number.MIN_SAFE_INTEGER;
    let minVal = Number.MAX_SAFE_INTEGER;
    let result = 0;
    let freq = new Array(26).fill(0);

    for(let i = 0 ; i < length ; i++)
    {
        for(let j = i ; j < length ; j++)
        {
            freq[s[j].charCodeAt(0) - 'a'.charCodeAt(0)]++;

            for(let el of freq)
            {
                if(el !== 0)
                {
                    maxVal = Math.max(maxVal, el);
                    minVal = Math.min(minVal, el);
                }
            }

            result += maxVal - minVal;
            maxVal = Number.MIN_SAFE_INTEGER;
            minVal = Number.MAX_SAFE_INTEGER;
        }

        freq.fill(0);
    }

    return result;
};